pkg/collector/annotations.go

```diff
@@ -37,13 +37,28 @@ func Annotations(instance v1alpha1.OpenTelemetryCollector) map[string]string {
 			annotations[k] = v
 		}
 	}
+	return annotations
+}
+
+func PodAnnotations(instance v1alpha1.OpenTelemetryCollector) map[string]string {
+	// new map every time, so that we don't touch the instance's annotations
+	annotations := map[string]string{}
+
+	if nil != instance.Spec.PodAnnotations {
+		for k, v := range instance.Spec.PodAnnotations {
+			annotations[k] = v
+		}
+	}
 	// make sure sha256 for configMap is always calculated
 	annotations["opentelemetry-operator-config/sha256"] = getConfigMapSHA(instance.Spec.Config)
 
 	return annotations
 }
 
 func getConfigMapSHA(config string) string {
-	h := sha256.Sum256([]byte(config))
-	return fmt.Sprintf("%x", h)
+	if config != "" {
```

Comment:
any reason why there is an empty check? 

what retunrs the hash func on an empty string?